src/test/HostActivationTests/NativeHosting/SharedTestStateBase.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.NativeHosting
+{
+    public class SharedTestStateBase : IDisposable
+    {
+        public string BaseDirectory { get; }
+        public string NativeHostPath { get; }
+        public RepoDirectoriesProvider RepoDirectories { get; }
+
+        public SharedTestStateBase()
+        {
+            BaseDirectory = SharedFramework.CalculateUniqueTestDirectory(Path.Combine(TestArtifact.TestArtifactsPath, "nativeHosting"));
+            Directory.CreateDirectory(BaseDirectory);
+
+            string nativeHostName = RuntimeInformationExtensions.GetExeFileNameForCurrentPlatform("nativehost");
+            NativeHostPath = Path.Combine(BaseDirectory, nativeHostName);
+
+            // Copy over native host and nethost
+            RepoDirectories = new RepoDirectoriesProvider();
+            string nethostName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("nethost");
```

Comment:
Nit: Why do we need the nethost copied into the comhost tests?

Reply:
Added linker option to nativehost to delay load nethost and moved copying nethost out of the base class.